internal/http/services/owncloud/ocs/handlers/apps/sharing/sharees/token.go

```diff
@@ -158,12 +154,11 @@ func handleBasicAuth(ctx context.Context, c gateway.GatewayAPIClient, token, pw
 	return c.Authenticate(ctx, &authenticateRequest)
 }
 
-func getTokenStatInfo(ctx context.Context, client gateway.GatewayAPIClient, token string) (*provider.StatResponse, error) {
-	return client.Stat(ctx, &provider.StatRequest{Ref: &provider.Reference{
-		ResourceId: &provider.ResourceId{
-			StorageId: utils.PublicStorageProviderID,
-			SpaceId:   utils.PublicStorageSpaceID,
-			OpaqueId:  token,
-		},
-	}})
+func getTokenStatInfo(ctx context.Context, client gateway.GatewayAPIClient, token string) (*link.GetPublicShareResponse, error) {
```

Comment:
I think it makes sense to update the name of this function since it doesn't return a `StatResponse` anymore.

Reply:
Good point :+1:  renamed to `getPublicShare` as this is literally what it does.